DokanNet/NtStatus.cs

```diff
@@ -109,359 +274,1663 @@ public enum NtStatus : long
+        /// </summary>
         TransactionInvalidMarshallBuffer = 0xc0190017,
+
+        /// <summary>
+        /// Error - The current transaction context associated with the thread
+        /// is not a valid handle to a transaction object.
+        /// </summary>
         CurrentTransactionNotValid = 0xc0190018,
+
+        /// <summary>
+        /// Error - An attempt to create space in the transactional resource
+        /// manager's log failed. The failure status has been recorded in the
+        /// event log.
+        /// </summary>
         LogGrowthFailed = 0xc0190019,
+
+        /// <summary>
+        /// Error - The object (file, stream, or link) that corresponds to the
+        /// handle has been deleted by a transaction save-point rollback.
+        /// </summary>
         ObjectNoLongerExists = 0xc0190021,
+
+        /// <summary>
+        /// Error - The specified file mini-version was not found for this
+        /// transacted file open.
+        /// </summary>
         StreamMiniversionNotFound = 0xc0190022,
+
+        /// <summary>
+        /// Error - The specified file mini-version was found but has been
+        /// invalidated. The most likely cause is a transaction save-point
+        /// rollback.
+        /// </summary>
         StreamMiniversionNotValid = 0xc0190023,
+
+        /// <summary>
+        /// Error - A mini-version can be opened only in the context of the
+        /// transaction that created it.
+        /// </summary>
         MiniversionInaccessibleFromSpecifiedTransaction = 0xc0190024,
+
+        /// <summary>
+        /// Error - It is not possible to open a mini-version with modify access.
+        /// </summary>
         CantOpenMiniversionWithModifyIntent = 0xc0190025,
+
+        /// <summary>
+        /// Error - It is not possible to create any more mini-versions for this
+        /// stream.
+        /// </summary>
         CantCreateMoreStreamMiniversions = 0xc0190026,
+
+        /// <summary>
+        /// Error - The handle has been invalidated by a transaction. The most
+        /// likely cause is the presence of memory mapping on a file or an open
+        /// handle when the transaction ended or rolled back to save-point.
+        /// </summary>
         HandleNoLongerValid = 0xc0190028,
+
+        /// <summary>
+        /// Error - There is no transaction metadata on the file.
+        /// </summary>
         NoTxfMetadata = 0xc0190029,
+
+        /// <summary>
+        /// Error - The log data is corrupt.
+        /// </summary>
+        /// <remarks>TODO: Is this the same as STATUS_NO_TXF_METADATA with value 0x80190029?</remarks>
         LogCorruptionDetected = 0xc0190030,
+
+        /// <summary>
+        /// Error - The file cannot be recovered because there is a handle still
+        /// open on it.
+        /// </summary>
+        /// <remarks>TODO: Is this the same as STATUS_CANT_RECOVER_WITH_HANDLE_OPEN with value 0x80190031?
```

Comment:
STATUS_CANT_RECOVER_WITH_HANDLE_OPEN: The file cannot be recovered because there is a handle still open on it.
https://msdn.microsoft.com/en-us/library/cc704588.aspx


From windows headers:
```
//
// MessageId: STATUS_CANT_RECOVER_WITH_HANDLE_OPEN
//
// MessageText:
//
// The file can't be recovered because there is a handle still open on it.
//
#define STATUS_CANT_RECOVER_WITH_HANDLE_OPEN ((NTSTATUS)0x80190031L)
```